realm/realm-library/src/main/java/io/realm/ProxyUtils.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm;
+
+import android.annotation.TargetApi;
+import android.os.Build;
+import android.util.JsonReader;
+import android.util.JsonToken;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+import io.realm.internal.OsList;
+import io.realm.internal.android.JsonUtils;
+
+class ProxyUtils {
+
+    /**
+     * Called by proxy to set the managed {@link RealmList} according to the given {@link JSONObject}.
+     *
+     * @param realmList the managed {@link RealmList}.
+     * @param jsonObject the {@link JSONObject} which may contain the data of the list to be set.
+     * @param fieldName the field name of the {@link RealmList}.
+     * @param <E> type of the {@link RealmList}.
+     * @throws JSONException if it fails to parse JSON.
+     */
+    static <E> void setRealmListWithJsonObject(
+            RealmList<E> realmList, JSONObject jsonObject, String fieldName) throws JSONException {
+        if (!jsonObject.has(fieldName))  {
+            return;
+        }
+
+        OsList osList = realmList.getOsList();
+        if (jsonObject.isNull(fieldName)) {
+            osList.removeAll();
+        }
+
+        JSONArray jsonArray = jsonObject.getJSONArray(fieldName);
+        realmList.getOsList().removeAll();
+        int arraySize = jsonArray.length();
+
+        if (realmList.clazz == Boolean.class) {
+            for (int i = 0; i < arraySize; i++) {
+                if (jsonArray.isNull(i)) {
+                    osList.addNull();
+                } else {
+                    osList.addBoolean(jsonArray.getBoolean(i));
+                }
+            }
+        } else if (realmList.clazz == Float.class) {
```

Comment:
Collapse this with below? 
`if (realmList.clazz == Double.class || realmList.claszz == Float.class)`

Reply:
`osList.addFloat((float) jsonArray.getDouble(i));` it needs to be casted to float